lib/pure/os.nim

```diff
@@ -3217,6 +3184,45 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo {.noWeirdTarget.
         raiseOSError(osLastError(), path)
     rawToFormalFileInfo(rawInfo, path, result)
 
+proc sameFileContent*(path1, path2: string; checkSize = false; bufferSize = 8192.Positive): bool {.
```

Comment:
test needed in tests/stdlib/tos.nim, where you can specify extreme values like `bufferSize=1` to check correctness there

Reply:
I think `bufferSize` must be a power of two ideally (?).

Reply:
what I mean is it's easier to test corner cases with small values of `bufferSize` (whether 1 or 2)

> must be a power of two ideally

why? man fread shows no such restriction

a test is needed, this function is sufficiently complex to justify it